automation/devops_automation_infra/utils/node_daemon.py

```diff
@@ -0,0 +1,42 @@
+import json
+import logging
+
+from devops_automation_infra.utils import consul
+from pipeng.utils import tracks
+from pipeng.services import pipe
+from devops_automation_infra.k8s_plugins.consul import Consul
+from devops_automation_infra.k8s_plugins.kafka import Kafka
+
+CONSUL_NAME = 'pipeng_local-agent-default'
+
+
+def create_kafka_tracks_reader(cluster, reader_offset='latest', topic="anv.tracks.pipeng.new-tracks"):
+    kafka_client = cluster.Kafka
+    tracks_reader = tracks.Reader(kafka_client, topic=topic, reader_offset=reader_offset)
+    return tracks_reader
+
+
+def get_list_pipes_from_consul(cluster):
+    consul_client = cluster.Consul.create_client()
+    instances = consul.get_service_nodes(consul_client, "pipeng")
+    logging.debug(f'Found {instances} in consul')
+    return instances
+
+
+def get_consul_path(layer, field, node_id=None):
+    return f"{layer}/{node_id}/{field}" if node_id is None else f"{layer}/{CONSUL_NAME}/{field}"
+
+
+def get_consul_value(cluster, key):
+    consul_client = cluster.Consul.create_client()
+
+    value_in_bytes = consul.get_key_layered(consul_client, key)
+    value_as_string = value_in_bytes.decode('utf-8')
+    try:
+        return json.loads(value_as_string)
+    except json.decoder.JSONDecodeError:
+        return value_as_string
```

Comment:
beautiful funtion.
what happens if value_in_bytes is none?
are you ok with simply returning value_as_string in case of a decoder error?

just some points to think about..

Reply:
thought about that and i think its better this way for a scenario which you insert nothin as a value and want to make sure its valid, thats the only possible reason, on the other hand there is an option of value coming back empty but it will probably be the up call stack responsibility to make sure what is the need from this value